libsolidity/analysis/PostTypeContractLevelChecker.cpp

```diff
@@ -112,18 +115,48 @@ void PostTypeContractLevelChecker::checkStorageLayoutSpecifier(ContractDefinitio
 		return;
 	}
 
-	if (rationalType->isFractional())
+	rational baseSlotRationalValue;
+	if (auto const integerType = dynamic_cast<IntegerType const*>(baseSlotExpressionType))
 	{
-		m_errorReporter.typeError(
-			1763_error,
-			baseSlotExpression.location(),
-			"The base slot of the storage layout must evaluate to an integer."
-		);
-		return;
+		if (integerType->isSigned())
```

Comment:
This could also be removed here and then replace the assert in line `173` as suggested in a previous PR https://github.com/ethereum/solidity/pull/15528#discussion_r1947107868

Reply:
It still fails the range check and the message is indeed clearer as you said:
```
Error: The base slot of the storage layout evaluates to -1, which is outside the range of type uint256.
```